src/eos-autoupdater.c

```diff
@@ -212,7 +213,8 @@ update_stamp_file (guint update_interval_days,
     return;
   }
 
-  g_get_current_time (&mtime);
+  mtime.tv_sec = last_successful_update_secs;
```

Comment:
What I meant here is that I understood that the `last_successful_update_secs` parameter tells us what the mtime of the timestamp should be, possibly altered a bit when the random delay is greater than zero. So we have 4 scenarios:
- `last_successful_update_secs` is now and random delay is 0 - timestamp of the file is now, so this is fine
- `last_successful_update_secs` is now and random delay is > 0 - timestamp is some random days in future
- `last_successful_update_secs` is 5 days ago and random delay is 0 - timestamp is now, so that's probably wrong
- `last_successful_update_secs` is 5 days ago and random delay is > 0 - timestamp is 5 days ago + random days, so that's fine.
(I'm not considering `last_successful_update_secs` being in future, but that's basically the same case as it being in the past.)

I'm actually predicting no problem at all here, since the third scenario does not happen at all so far - in this situation we perform the update immediately. It just looked wrong and likely not futureproof if we ever change the update logic again.